### PR TITLE
use C10_REGISTER for GELU op

### PR DESCRIPTION
**Pull request by hl475**
Differential Revision: D14864737

### CODE REVIEW

**Review by houseroad — CHANGES_REQUESTED** [review]
Please add the test here: https://github.com/pytorch/pytorch/blob/master/caffe2/python/operator_test/torch_integration_test.py

File: caffe2/operators/gelu_op.cc [hunk 1]
```diff
@@ -130,3 +122,12 @@ class GetGeluGradient : public GradientMakerBase {
 REGISTER_GRADIENT(Gelu, GetGeluGradient);
 
 } // namespace caffe2
+
+C10_REGISTER_CAFFE2_OPERATOR_CPU(
+    Gelu,
+    (std::vector<c10::Argument>{
+        c10::Argument("input"),
+        c10::Argument("fast_gelu", BoolType::get()),
```
Comment (houseroad) [inline caffe2/operators/gelu_op.cc]: set default value?

File: caffe2/operators/gelu_op.cc [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #include <string>
 #include <vector>
 
+#include <c10/core/Tensor.h>
```
Comment (houseroad) [inline caffe2/operators/gelu_op.cc]: why include Tensor.h?

File: caffe2/python/operator_test/torch_integration_test.py [hunk 3]
```diff
@@ -447,6 +448,26 @@ def test_roi_align_cpu(self):
...
+    def _test_gelu_op(self, X, fast_gelu, device):
+        def _gelu_ref(_X):
+            return (_X * norm.cdf(_X).astype(np.float32), )
+        expected_output, = _gelu_ref(X)
+        actual_output = torch.ops._caffe2.Gelu(torch.tensor(X), fast_gelu)
+
+        rtol = 1e-3 if fast_gelu else 1e-4
+        atol = 1e-5
+        torch.testing.assert_allclose(
+            expected_output, actual_output, rtol=rtol, atol=atol)
```
Comment (houseroad) [inline caffe2/python/operator_test/torch_integration_test.py]: actual_output.cpu()

**Review by houseroad — COMMENTED** [review]
Please address inline comments

**Review by houseroad — APPROVED** [review]
Looks good.

**Comment by houseroad** [comment]
Let's wait a bit for the IC signals

**Review by xiaomengy — APPROVED** [review]

**Pull request closed without merging**